lib/Common/PlatformAgnostic/SystemInfo.h

```diff
@@ -33,6 +33,50 @@ namespace PlatformAgnostic
         }
 
         static bool GetMaxVirtualMemory(size_t *totalAS);
+
+#define SET_BINARY_PATH_ERROR_MESSAGE(path, msg) \
+    str_len = (int) strlen(msg);                 \
+    memcpy(path, msg, (size_t)str_len);          \
+    path[str_len] = char(0)
+
+#ifdef _WIN32
+        static void GetBinaryLocation(char *path, const unsigned size)
+        {
+            AssertMsg(path != nullptr, "Path can not be nullptr");
```

Comment:
The reason for the errors. AssertMsg is not defined here. Why don't you comment out here and add `// TODO: make AssertMsg available under PlatformAgnostic` ?